### PR TITLE
Upgrade to React 18

### PR DESCRIPTION
**Pull request by Sergio-Mira**
Upgrade to React 18.

## Which problem is this PR solving?
- Resolves https://github.com/jaegertracing/jaeger-ui/issues/998

## Short description of the changes
- Upgrade to React 18

### CODE REVIEW

File: packages/jaeger-ui/package.json [hunk 1]
```diff
@@ -85,10 +85,10 @@
     "prop-types": "^15.5.10",
     "query-string": "^6.3.0",
     "raven-js": "^3.22.1",
-    "react": "^16.14.0",
-    "react-circular-progressbar": "^2.0.3",
-    "react-dimensions": "^1.3.0",
-    "react-dom": "^16.14.0",
+    "react": "^18.2.0",
```
Comment (yurishkuro) [inline packages/jaeger-ui/package.json:88]: Wow, that looks suspiciously easy, were there really no breaking changes?

Comment (Sergio-Mira) [reply packages/jaeger-ui/package.json:88]: The breaking changes of lifecycle etc have been postponed to upcoming versions of React 18 and are still warnings afaik, and that is why its an easier upgrade. As far as I understand and can test the project still can use dependencies which are not yet fully React 18 compatible (like react-vis in this case).

Would love to hear how to move this PR forward and what we can do to test it, what edge cases of libs we think might break.

(I've also submitted changes to migrate from old lifecycle events in react-vis here https://github.com/uber/react-vis/pull/1473)

File: packages/jaeger-ui/src/setupTests.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ rafPolyfill();
 
 /* eslint-disable import/no-extraneous-dependencies */
 const Enzyme = require('enzyme');
-const EnzymeAdapter = require('enzyme-adapter-react-16');
+const EnzymeAdapter = require('@wojtekmaj/enzyme-adapter-react-17');
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/setupTests.js:23]: I can see this is another future PITA to be solved -- https://testing-library.com/docs/react-testing-library/migrate-from-enzyme/

**Review by yurishkuro — APPROVED** [review]
did a brief test of different views, don't see any immediate issues, so let's give it a try

**Comment by metasal1** [comment]
Any plans to merge?

**Comment by yurishkuro** [comment]
@metasal1 merge what? This PR was merged in Feb
